lesson17/src/js/main.js

```diff
@@ -0,0 +1,217 @@
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.classList.add("arrow", "prev");
+  prevButton.id = "js-button_prev";
+  prevButton.setAttribute("aria-label", "previous");
+  prevButton.textContent = "previous";
+
+  fragment.appendChild(prevButton);
+
+  const nextButton = document.createElement("button");
+  nextButton.classList.add("arrow", "next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const slideArea = document.querySelector(".mainvisual__images");
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = document.createElement("div");
+  pagenation.classList.add("slider__pagination");
+
+  let fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.classList.add("current");
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = document.createElement("span");
+  separation.textContent = "/";
+  separation.classList.add("separation");
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.classList.add("total");
+  total.textContent = item.length;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function manipulatePagination(target) {
+  const pagenationCurrent = document.querySelector(".current");
+  const pagenationTotal = Number(document.querySelector(".total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  const buttonPrev = target.getAttribute("aria-label") == 'previous';
+  const buttonNext = target.getAttribute("aria-label") == 'next';
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum +=1;
+  }else if(currentNum <= pagenationTotal && buttonPrev){
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function ImageNextSlide(e) {
+  const slideImage = document.querySelectorAll(".slider__item");
```

Comment:
nits
It can be written this way.
Also, I thought it would be easier to understand if the variable names were plural like images

```suggestion
  const slideImages = [...document.getElementsByClassName(".slider__item")];
```

Reply:
It's easier to operate later if you expand it with spread syntax!
This has been modified along with the variable name.

[16f6958](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/16f69581d2e55c8f75334773ed65e61355a1d8b4)